fri/src/verifier.rs

```diff
@@ -86,65 +95,26 @@ pub(crate) fn verify<FC: FriConfig>(
     Ok(())
 }
 
-fn verify_input<FC: FriConfig>(
-    input_mmcs: &[FC::InputMmcs],
-    input_commits: &[<FC::InputMmcs as Mmcs<FC::Val>>::Commitment],
-    input_dims: &[Vec<Dimensions>],
-    input_openings: &[InputOpening<FC>],
-    index: usize,
-    alpha: FC::Challenge,
-    log_max_height: usize,
-) -> VerificationResult<FC, Vec<FC::Challenge>> {
-    let mut openings_by_log_height: Vec<Vec<FC::Val>> = vec![vec![]; log_max_height + 1];
-    for (mmcs, commit, dims, opening) in
-        izip!(input_mmcs, input_commits, input_dims, input_openings)
-    {
-        mmcs.verify_batch(
-            commit,
-            dims,
-            index,
-            &opening.opened_values,
-            &opening.opening_proof,
-        )
-        .map_err(VerificationError::InputMmcsError)?;
-        for (mat_dims, mat_opening) in izip!(dims, &opening.opened_values) {
-            let log_height = log2_strict_usize(mat_dims.height);
-            openings_by_log_height[log_height].extend_from_slice(mat_opening);
-        }
-    }
-    let reduced_openings = openings_by_log_height
-        .into_iter()
-        .map(|o| {
-            o.into_iter()
-                .zip(alpha.powers())
-                .map(|(y, alpha_pow)| alpha_pow * y)
-                .sum()
-        })
-        .collect();
-    Ok(reduced_openings)
-}
-
 fn verify_query<FC: FriConfig>(
     config: &FC,
     commit_phase_commits: &[<FC::CommitPhaseMmcs as Mmcs<FC::Challenge>>::Commitment],
     mut index: usize,
     proof: &QueryProof<FC>,
     betas: &[FC::Challenge],
-    reduced_openings: Vec<FC::Challenge>,
+    reduced_openings: &[FC::Challenge; 32],
```

Comment:
I use `[usize; 32]` and `[F; 32]` in a few places, where 32 is the max log height of a matrix (ever). So it's assuming we won't have 2^32 blown up rows. Then the info for a given `log_height` is just at `arr[log_height]`.

Kind of weird, but it feels nicer than a bunch of Vecs that can get mixed up with varying lengths? Lmk if too weird

Reply:
Hm not that weird but I don't quite see the appeal, seems like there's still an implicit "actual" length, making it similar to `Vec::with_capacity(32)`?

Reply:
I guess the nicer aspect of `[_; 32]` is that we don't need to check if the `Vec` is too small and needs to be expanded. I think ideally we'd have a little data structure (something like `IndexMap`?) which takes care of this, but `[_; 32]` seems fine for now.